sdk/monitor/Azure.Monitor.Ingestion/src/Models/UploadLogsOptions.cs

```diff
@@ -21,5 +28,18 @@ public class UploadLogsOptions
         /// <remarks> In the Upload method, this parameter is not used as the batches are uploaded in sequence. For parallel uploads, if this value is not set the default concurrency will be 5. </remarks>
         /// </summary>
         public int MaxConcurrency { get; set; }
+
+        /// <summary>
+        /// test
+        /// </summary>
+        public event SyncAsyncEventHandler<UploadFailedArgs> UploadFailed;
+
+        private ClientDiagnostics _clientDiagnostics = new ClientDiagnostics(new LogsIngestionClientOptions());
```

Comment:
This should be the same clientDiagnostics isntance used by the client.

Reply:
I followed this example where the ClientDiagnostics are also made in the same class. I don't think I can pass in the same ClientDiagnostics between classes.

Reply:
Why can't we pass it? The idea is that we want to use the diagnostics based on the user specified options. Instead we are creating a new diagnostics instance that ignores their settings.

Reply:
Yes, we should pass the client's into the `InvokeEvent` method.